python/src/main.cpp

```diff
@@ -131,34 +137,32 @@ PYBIND11_MODULE(kp, m)
       m, "Tensor", DOC(kp, Tensor))
       .def(
         "data",
-        [](kp::Tensor& self) {
+        [np](kp::Tensor& self) {
             // Non-owning container exposing the underlying pointer
-            switch (self.dataType()) {
-                case kp::Tensor::TensorDataTypes::eFloat:
-                    return py::array(
-                      self.size(), self.data<float>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eUnsignedInt:
-                    return py::array(
-                      self.size(), self.data<uint32_t>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eInt:
-                    return py::array(
-                      self.size(), self.data<int32_t>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eDouble:
-                    return py::array(
-                      self.size(), self.data<double>(), py::cast(&self));
-                case kp::Tensor::TensorDataTypes::eBool:
-                    return py::array(
-                      self.size(), self.data<bool>(), py::cast(&self));
-                default:
-                    throw std::runtime_error(
-                      "Kompute Python data type not supported");
-            }
+            auto frombuffer = np.attr("frombuffer");
```

Comment:
Don't use auto in source, only in examples for reader simpliticy - explicit is better

Reply:
will change

Reply:
changed